<!--REQUEST_ID=434bd29b-f550-491c-8237-d19eaa580ca2-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.50506&attempt_id=47c3733e-e888-452b-9c82-75dc720bfca1)

Attempted: stress-ng

<details><summary>Partial log (click to expand)</summary><p>

```
Undefined symbols for architecture x86_64:
  "_getentropy", referenced from:
      _shim_getrandom in shim.o
ld: symbol(s) not found for architecture x86_64
clang-5.0: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile:350: stress-ng] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-stress-ng-0.09.46.drv-0/stress-ng-0.09.46'
make: *** [Makefile:337: all] Error 2
builder for '/nix/store/lr7893vvqp9idq95xfwldmv07ll9gki7-stress-ng-0.09.46.drv' failed with exit code 2
error: build of '/nix/store/lr7893vvqp9idq95xfwldmv07ll9gki7-stress-ng-0.09.46.drv' failed
```
</p></details>

--- a/pkgs/tools/system/stress-ng/default.nix
+++ b/pkgs/tools/system/stress-ng/default.nix
@@ -1,15 +1,21 @@
-{ stdenv, fetchurl, attr, keyutils }:
+{ stdenv, fetchurl
+, attr, keyutils, libaio, libapparmor, libbsd, libcap, libgcrypt, lksctp-tools, zlib
+}:
 
 stdenv.mkDerivation rec {
   name = "stress-ng-${version}";
-  version = "0.06.14";
+  version = "0.09.46";
 
   src = fetchurl {
-    sha256 = "06kycxfwkdrm2vs9xk8cb6c1mki29ymrrqwwxxqx4icnwvq135hv";
-    url = "http://kernel.ubuntu.com/~cking/tarballs/stress-ng/${name}.tar.gz";
+    url = "http://kernel.ubuntu.com/~cking/tarballs/stress-ng/${name}.tar.xz";
+    sha256 = "0m1f46vqixx2mgqdrjwkl8w9did7n99sy96rbcgkkn9g99y59qjm";
   };
 
-  buildInputs = [ attr keyutils ];
+  # All platforms inputs then Linux-only ones
+  buildInputs = [ libbsd libgcrypt zlib ]
+    ++ stdenv.lib.optionals stdenv.hostPlatform.isLinux [
+      attr keyutils libaio libapparmor libcap lksctp-tools
+    ];
 
   patchPhase = ''
     substituteInPlace Makefile --replace "/usr" ""
@@ -36,9 +42,10 @@ stdenv.mkDerivation rec {
       hardware issues such as thermal overruns as well as operating system
       bugs that only occur when a system is being thrashed hard.
     '';
-    homepage = http://kernel.ubuntu.com/~cking/stress-ng;
+    homepage = http://kernel.ubuntu.com/~cking/stress-ng/;
     downloadPage = http://kernel.ubuntu.com/~cking/tarballs/stress-ng/;
     license = licenses.gpl2Plus;
-    platforms = platforms.linux;
+    maintainers = with maintainers; [ c0bw3b ];
+    platforms = platforms.linux; # TODO: fix https://github.com/NixOS/nixpkgs/pull/50506#issuecomment-439635963
   };
 }

